tools/project.py

```diff
@@ -898,6 +946,34 @@ def write(self, n: ninja_syntax.Writer) -> None:
         source_inputs: List[Path] = []
         source_added: Set[Path] = set()
 
+        all_pch_outputs = []
+        if config.precompiled_headers:
+            for pch in config.precompiled_headers:
+                pch_out_name = Path(pch["source"]).with_suffix(".mch")
+
+                n.comment(f"Precompiled header {pch_out_name}")
+                pch_out_abs_path = Path(get_pch_out_name(config, pch))
+                n.build(
+                    outputs=pch_out_abs_path,
+                    rule="mwcc_pch",
+                    inputs=f"include/{pch["source"]}",
+                    variables={
+                        "mw_version": Path(pch["mw_version"]),
+                        "cflags": make_flags_str(pch["cflags"]),
```

Comment:
I wonder if we need the automatic `-lang` insertion here too (see `def c_build`)

Reply:
I think this should be implemented now, let me know if it looks okay.

Reply:
Looks good!